### ISSUE TITLE
How to get keypoints output .json file like original OpenPose ?

### ISSUE THREAD

**Issue by mayank64ce** [patrickvonplaten/controlnet_aux#83]

**Comment by patrickvonplaten** [comment]
Sorry I don't understand the question

**Comment by mayank64ce** [comment]
I meant the output of the original OpenPose is a json file with 2d keypoints which look something like this:

`{
  "version": 1.3,
  "people": [
    {
      "person_id": [
        -1
      ],
      "pose_keypoints_2d": [
        535.734,
        246.827,
        0.855832,
        ....
      ],
      "face_keypoints_2d": [
        496.62,
        230.471,
        0.699601,
        ...
      ],
      "hand_left_keypoints_2d": [
        751.966,
        512.104,
        0.542437,
        ...
      ],
      "hand_right_keypoints_2d": [
        266.106,
        510.039,
        0.540713,
        ....
      ],
      "pose_keypoints_3d": [],
      "face_keypoints_3d": [],
      "hand_left_keypoints_3d": [],
      "hand_right_keypoints_3d": []
    }
  ]
}`

Can I get these values from controlnet_aux ?

**Comment by pdoane** [comment]
This functionality is in the A1111 implementation, e.g. see https://github.com/Mikubill/sd-webui-controlnet/blob/main/annotator/openpose/__init__.py#L122

**Comment by mayank64ce** [comment]
So if I pass the `poses` from https://github.com/patrickvonplaten/controlnet_aux/blob/master/src/controlnet_aux/open_pose/__init__.py#L220

to that function, I should be able to get the json right ?

**Comment by pdoane** [comment]
There's a good chance that the openpose implementations have diverged.  I last updated all of them on 5/27 so it's worth a pass to review.

**Comment by mayank64ce** [comment]
yeah latest openpose has 7 additinal joints. The pelvis + 2 toes and ankle for both feet. Anyways those didn't matter to me. I approximated the position of pelvis with hip joints and mapped the toes and ankle to feet.
